packages/govuk-frontend/src/govuk/init.jsdom.test.mjs

```diff
@@ -292,6 +294,49 @@ describe('createAll', () => {
     )
   })
 
+  it('executes overloaded checkSupport of component', () => {
+    const componentRoot = document.createElement('div')
+    componentRoot.setAttribute('data-module', 'mock-component')
+    document.body.appendChild(componentRoot)
+
+    const checkSupportMock = jest.fn(() => {})
+
+    class MockComponentWithCheckSupport extends MockComponent {
+      checkSupport() {
+        checkSupportMock()
+      }
+    }
```

Comment:
**suggestion** Possibly a long shot, but does [`jest.spyOn`](https://jestjs.io/docs/jest-object#jestspyonobject-methodname) let us avoid creating one class per test with the method being static?

```js
const checkSupportMock = jest.spyOn(MockComponent, 'checkSupport')
```